olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -426,13 +443,14 @@ HttpResponse OlpClient::OlpClientImpl::CallApi(
           .WithProxySettings(
               settings_.proxy_settings.value_or(http::NetworkProxySettings()));
 
-  auto network_request =
-      http::NetworkRequest(utils::Url::Construct(base_url_, path, query_params))
-          .WithVerb(GetHttpVerb(method))
-          .WithBody(std::move(post_body))
-          .WithSettings(std::move(network_settings));
+  auto network_request = http::NetworkRequest(
+      utils::Url::Construct(base_url_, path, query_params));
+
+  AddBearer(path, query_params.empty(), network_request);
 
-  AddBearer(settings_.authentication_settings, network_request);
+  network_request.WithVerb(GetHttpVerb(method))
+      .WithBody(std::move(post_body))
+      .WithSettings(std::move(network_settings));
```

Comment:
Why is it important for these to be set after AddBearer()? I don't see any connection.